src/Content/Version.php

```diff
@@ -146,23 +165,88 @@ public function move(
 		$this->ensure($fromLanguage);
 		$this->model->storage()->move(
 			fromVersionId: $this->id,
-			fromLanguage: $this->language($fromLanguage),
+			fromLanguage: Language::ensure($fromLanguage),
 			toVersionId: $toVersionId,
-			toLanguage: $this->language($toLanguage)
+			toLanguage: Language::ensure($toLanguage)
 		);
 	}
 
 	/**
 	 * Returns the stored content fields
 	 *
 	 * @return array<string, string>
+	 */
+	public function read(Language|string $language = 'default'): array
+	{
+		$language = Language::ensure($language);
+
+		try {
+			$fields = $this->model->storage()->read($this->id, $language);
+			$fields = $this->convertFieldNamesToLowerCase($fields);
+			return $fields;
+		} catch (Throwable) {
+			return [];
```

Comment:
`Question`: Do we need to return an empty array here? As I wrote before, I feel like this could fall on our feet as it becomes impossible to differentiate between content that does not exist at all and empty content. If possible, I'd prefer `null` for clarity.

Reply:
My main reason to go for the array was to simplify the usage of ::read in other places. We already have `Version::exists` to check if the version is actually there. That's why I thought we already offer ways to detect this properly.

Reply:
About simplification of calling code: With `null`, we could use `?? []` where we need it. This would also make the behavior more explicit and visible at the first glance.

An argument why `exists()` is not enough: We could run into race conditions where a version first exists when `exists()` is called but is then deleted before `read()` is called. So the calling code would get an empty array and think it's the actual content, even though the version was deleted. If we directly call `read()` and check on `null`, we can be sure that the data is consistent. The race condition will be a rare edge case, but it will be even harder to debug if it ever happens. So IMO it would be good to avoid it from the start.

So TL;DR: With `null` returned on non-existing versions, we can improve reliability and increase flexibility for the calling code while making the calling code only slightly longer.

Reply:
Ok, convinced :) I will try to adapt the code in those places where the array is expected.